File: swimming-contest-software/Model/src/main/java/domain/enums/SwimmingStyles.java
package domain.enums;

/**
 * Available swimming styles for races.
 */
public enum SwimmingStyles {
    _MIXED,
    _FREESTYLE,
    _BUTTERFLY,
    _BACKSTROKE;

    public static SwimmingStyles styleFromInteger(Integer value) {
        return switch (value) {
            case 1 -> _MIXED;
            case 2 -> _FREESTYLE;
            case 3 -> _BUTTERFLY;
            case 4 -> _BACKSTROKE;
            default -> null;
        };
    }

    public static Integer integerFromStyle(SwimmingStyles swimmingDistance) {
        return switch (swimmingDistance) {
            case _MIXED -> 1;
            case _FREESTYLE -> 2;
            case _BUTTERFLY -> 3;
            case _BACKSTROKE -> 4;
        };
    }
}

File: Model/Domain/Enums/SwimmingStyles.cs
using System;

namespace Model.Domain.Enums;

public enum SwimmingStyles
{
    Mixed,
    Freestyle,
    Butterfly,
    Backstroke
}

public static class SwimmingStylesMethods
{
    public static SwimmingStyles StyleFromInteger(Int32 value) =>
        value switch
        {
            1 => SwimmingStyles.Mixed,
            2 => SwimmingStyles.Freestyle,
            3 => SwimmingStyles.Butterfly,
            4 => SwimmingStyles.Backstroke,
            _ => throw new ArgumentOutOfRangeException(nameof(value), value, null)
        };

    public static Int32 IntegerFromStyle(SwimmingStyles style) =>
        style switch
        {
            SwimmingStyles.Mixed => 1,
            SwimmingStyles.Freestyle => 2,
            SwimmingStyles.Butterfly => 3,
            SwimmingStyles.Backstroke => 4,
            _ => throw new ArgumentOutOfRangeException(nameof(style), style, null)
        };
}

